### PR TITLE
Governance 2.0

### PR DESCRIPTION
**Pull request by jameshowlett977**
What should be done:
- [x] merge code from PR `Development Fund Contract and Testing`
- [x]  merge code from PR `SOV.approveAndCall`
- [x]  add function `delegate` to Vesting
- [x]  add inline documentation
- [x]  update GovernorAlpha.votingPeriod, Timelock.delay

### CODE REVIEW

File: contracts/governance/GovernorAlpha.sol [hunk 1]
```diff
@@ -398,11 +398,12 @@ contract GovernorAlpha is SafeMath96 {
 			return ProposalState.Active;
 		}
 
+		uint96 totalVotes = add96(proposal.forVotes, proposal.againstVotes, "GovernorAlpha:: state: forVotes + againstVotes > uint96");
+		uint96 totalVotesMajorityPercentage = div96(totalVotes, 100, "GovernorAlpha:: state: division error");
```
[the remark this answers is not in the document]
Comment (ororopickpocket) [reply contracts/governance/GovernorAlpha.sol:402]: this is tricky :)
in all usual cases you'd be right. but we're limited to a 96 bit representation and with the voting weight we're getting close to the max. a multiplication by 100 can easily cause an overflow.

**Review by powerhousefrank — COMMENTED** [review]
The schedule looks good to me :+1:

**Review by powerhousefrank — COMMENTED** [review]
Change verified in the schedule :+1:

File: scripts/contractInteraction/testnet_contracts.json [hunk 2]
```diff
@@ -23,6 +23,11 @@
     "BPro": "0x4dA7997A819bb46B6758b9102234c289Dd2ad3bf",
     "RSKOracle": "0xE00243Bc6912BF148302e8478996c98c22fE8739",
     "PriceFeedRSKOracle":"0xF2B7440C89431DF82EC6c8F3D079059847565dF0",
+
+    "cSOV1": "0x75bbf7f4d77777730eE35b94881B898113a93124",
```
Comment (powerhousefrank) [inline scripts/contractInteraction/testnet_contracts.json:27]: Based on the new commit b78b219, better to change it here as well. And the next line.

File: scripts/contractInteraction/mainnet_contracts.json [hunk 3]
```diff
@@ -27,5 +27,21 @@
 
     "CSOV1": "0x0106F2fFBF6A4f5DEcE323d20E16E2037E732790",
     "CSOV2": "0x7f7Dcf9DF951C4A332740e9a125720DA242A34ff",
-    "governorVault": "0xC7A1637b37190a456b017897207bceb2A29f19b9"
+    "governorVault": "0xC7A1637b37190a456b017897207bceb2A29f19b9",
+
+    "SOV": "0xEFc78fc7d48b64958315949279Ba181c2114ABBd",
```
Comment (powerhousefrank) [inline scripts/contractInteraction/mainnet_contracts.json:32]: Should we make it checksum valid?

https://explorer.rsk.co/check-address/0xEFc78fc7d48b64958315949279Ba181c2114ABBd
vs
https://explorer.rsk.co/check-address/0xefc78fc7d48b64958315949279ba181c2114abbd

**Pull request merged**